### DISCUSSION TITLE
Why not remove health-related fields from FlxObject/FlxBasic?

### DISCUSSION THREAD

**Discussion by bogdan-ov** [HaxeFlixel/flixel#2862] [Ideas]
Which fields do I propose to remove from `FlxObject` and `FlxBasic`:
- `object.health`
- `object.hurt()`
- `basic.alive`
- `basic.kill()`
- `basic.revive()`

For example, I want the `health` parameter to be of the `Int` type, but it is already occupied under the `Float` type. Another example: I want to create a method like `hurt(damage:Int, knockback:Float)`, but this name is already taken and I can't add or change the arguments of the method, so I need to create other method with ugly name like `takeDamage(damage:Int, knockback:Float)`. It's inconvenient
I also think that the `alive` parameter is pretty useless, because it won't be difficult for me to add custom `alive` parameter and the `kill()` method myself (I may also need custom arguments in the `kill()` method)

If someone really needs these fields, then it seems to me that it will not be difficult for them to add them themselves (each field takes literally a couple lines of code) when switching to a new version of the engine

**I may be wrong and this is purely my opinion, so I would like to hear yours :)**

**Comment by Geokureli** [comment]
The main reason is backwards compatibility. we don't really know how many flixel use this feature and how many projects would be broken if this was removed. That said, I agree, health shouldn't be there.

> If someone really needs these fields, then it seems to me that it will not be difficult for them to add them themselves (each field takes literally a couple lines of code) when switching to a new version of the engine

I'm still dealing with people who haven't fixed small deprecation warnings introduced in 4.10. breaking changes will stop a lot of people in their tracks. we can only introduce breaking changes at major version releases and if we do it too often people just won't upgrade. Specifically removing the `alive` field alone would break all of my projects, it's pretty commonly used for something that you want drawn and updating, but not in an active state

**Reply by bogdan-ov** [reply]
Thanks for the answer! I think I'll just delete these fields from the engine code

**Reply by Geokureli** [reply]
Please don't
<img width="590" alt="Screenshot 2023-07-21 at 9 44 28 AM" src="https://github.com/HaxeFlixel/flixel/assets/2609513/303cebef-47ce-4853-95d1-524715367853">

**Reply by Geokureli** [reply]
whoever started this trend of "just edit the source" is my most hated enemy, this is absolutely unheard of in all other areas of open source dev

**Reply by Geokureli** [reply]
just use different field names than `health` and `hurt` for crying out loud

**Reply by bogdan-ov** [reply]
I know it's very bad to change the library code directly, but I can't resist. Looks like I'll have to find another way. Thanks again

**Reply by Geokureli** [reply]
FYI this also goes for making a copy of flixel/FlxSprite.hx in your repo and using that
